database/migrations/2019_09_28_171753_config_fix.php

```diff
@@ -0,0 +1,577 @@
+				$config->value = $value['value'];
+				$config->save();
+			}
+		}
+	}
+
+	/**
+	 * Run the migrations.
+	 *
+	 * @return void
+	 */
+	public function up()
+	{
+		define('INT', 'int');
+		define('STRING', 'string');
+		define('STRING_REQ', 'string_required');
+		define('BOOL', '0|1');
+		define('TERNARY', '0|1|2');
+		define('DISABLED', '');
+
+		$default_values = [
+			[
+				'key' => 'version',
+				'value' => '040000',
+				'cat' => 'Admin',
+				'type_range' => INT,
+				'confidentiality' => '0',
+			],
+			[
+				'key' => 'username',
+				'value' => '',
+				'cat' => 'Admin',
+				'type_range' => STRING_REQ,
+				'confidentiality' => '4',
+			],
+			[
+				'key' => 'password',
+				'value' => '',
+				'cat' => 'Admin',
+				'type_range' => STRING_REQ,
+				'confidentiality' => '4',
+			],
+			[
+				'key' => 'check_for_updates',
+				'value' => '0',
+				'cat' => 'Admin',
+				'type_range' => BOOL,
+				'confidentiality' => '0',
+			],
+			[
+				'key' => 'sorting_Photos_col',
+				'value' => 'title',
+				'cat' => 'Gallery',
+				'type_range' => 'id|takestamp|title|description|public|star|type',
+				'confidentiality' => '2',
+			],
+			[
+				'key' => 'sorting_Photos_order',
+				'value' => 'ASC',
+				'cat' => 'Gallery',
+				'type_range' => 'ASC|DESC',
+				'confidentiality' => '2',
+			],
+			[
+				'key' => 'sorting_Albums_col',
+				'value' => 'max_takestamp',
+				'cat' => 'Gallery',
+				'type_range' => 'id|title|description|public|max_takestamp|min_takestamp',
+				'confidentiality' => '2',
+			],
+			[
+				'key' => 'sorting_Albums_order',
+				'value' => 'ASC',
+				'cat' => 'Gallery',
+				'type_range' => 'ASC|DESC',
+				'confidentiality' => '2',
+			],
+			[
+				'key' => 'imagick',
+				'value' => '1',
+				'cat' => 'Image Processing',
+				'type_range' => BOOL,
+				'confidentiality' => '2',
+			],
+			[
+				'key' => 'dropbox_key',
+				'value' => '',
+				'cat' => 'Admin',
+				'type_range' => STRING,
+				'confidentiality' => '3',
+			],
+			[
+				'key' => 'skip_duplicates',
+				'value' => '0',
+				'cat' => 'Image Processing',
+				'type_range' => BOOL,
+				'confidentiality' => '2',
+			],
+			[
+				'key' => 'lang',
+				'value' => 'en',
+				'cat' => 'Gallery',
+				'type_range' => DISABLED,
+				'confidentiality' => '0',
+			],
+			[
+				'key' => 'layout',
+				'value' => '1',
+				'cat' => 'Gallery',
+				'type_range' => TERNARY,
+				'confidentiality' => '0',
+			],
+			[
+				'key' => 'image_overlay',
+				'value' => '1',
+				'cat' => 'Gallery',
+				'type_range' => BOOL,
+				'confidentiality' => '0',
+			],
+			[
+				'key' => 'default_license',
+				'value' => 'none',
+				'cat' => 'Gallery',
+				'type_range' => STRING_REQ,
+				'confidentiality' => '2',
+			],
+			[
+				'key' => 'small_max_width',
+				'value' => '0',
+				'cat' => 'Image Processing',
+				'type_range' => INT,
+				'confidentiality' => '2',
+			],
+			[
+				'key' => 'small_max_height',
+				'value' => '360',
```

Comment:
I meant to ask this since forever: why does this default to `360`? justified-layout's `targetRowHeight` defaults to `320`, though as far as I can tell, the height is allowed to vary from half to twice as much, i.e., from `160` to `640`. Given that, `360` seems so... arbitrary. To ensure high quality, it should probably default to `640`...

Reply:
My guess would be that it's half of 720 (as lots of photos and screens are build around HD-type resolutions now)